purrsoft/src/scenes/Voluntari/component.tsx

```diff
@@ -0,0 +1,177 @@
+
+export const Voluntari = () => {
+  const [searchTerm, setSearchTerm] = useState('');
+  const { data, error, isLoading } = useGetVolunteersQuery();
+
+  const theme = useTheme();
+
+  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchTerm(event.target.value.toLowerCase());
+  };
+
+  const filteredVolunteers = 
+    data?.records.filter(
+        (volunteer) => 
+            (volunteer.firstName.toLowerCase() + ' ' +
+            volunteer.lastName.toLowerCase()).includes(searchTerm)
+    ) || [];
+
+  if (isLoading) {
+    return <div>Loading...</div>;
+  }
+
+  if (error) {
+    return <div>Error fetching volunteers!</div>;
+  }
+
+  return (
+    <Container>
+      {/* Sticky Search Bar */}
+      <Box
+        sx={{
+          position: 'sticky',
+          top: 0,
+          backgroundColor: theme.palette.background.default,
+          zIndex: 9,
+          py: 1,
+        }}
+      >
+        <Box sx={{ display: 'flex', justifyContent: 'left', my: 2 }}>
+          <TextField
+            variant="outlined"
+            placeholder="Caută voluntari"
+            value={searchTerm}
+            onChange={handleSearchChange}
+            sx={{
+              width: '100%',
+              maxWidth: '600px',
+              borderRadius: '50px',
+              '& .MuiOutlinedInput-root': {
+                borderRadius: '50px',
+              },
+            }}
+            InputProps={{
+              startAdornment: (
+                <InputAdornment position="start">
+                  <IconButton>
+                    <Search />
+                  </IconButton>
+                </InputAdornment>
+              ),
+              endAdornment: (
+                <InputAdornment position="end">
+                  <IconButton>
+                    <Mic />
+                  </IconButton>
+                </InputAdornment>
+              ),
+            }}
+          />
+        </Box>
+      </Box>
+
+      {/* Volunteer Cards */}
+      <Box sx={{ ml: 2, mr: 1, pb: 4 }}>
+        <Grid container spacing={4}>
+          {filteredVolunteers.length > 0 ? (
+            filteredVolunteers.map((volunteer) => (
+            <Grid item xs={12} sm={6} md={4} lg={3} key={volunteer.userId}>
+              <CustomCard
+                width="100%"
+                height="100%"
+                align="center"
+                justify="center"
+              >
+                <Box sx={{ textAlign: 'center', py: 2 }}>
+                  <Typography
+                    sx={{
+                      backgroundColor: theme.palette.accent?.lightGreen,
+                      borderRadius: '8px',
+                      mb: 1,
+                      color: theme.palette.accent?.white,
+                      fontSize: '18px',
+                      padding: '6px',
+                    }}
+                  >
+                    Nume: {volunteer.firstName} {volunteer.lastName}
```

Comment:
Ok so I may be out of the loop with things but wasn't this supposed to be the second item of the grid? Where is the username. Just asking 'cause I am confused.

Reply:
A user does not have a username. Sabina said that when she made the design, she put there some fields to see how the card should look like, so I decided to show the name, status and tier only, but I could display the email instead of username.